var/spack/repos/builtin/packages/ImageMagick/package.py

```diff
@@ -42,6 +43,9 @@ class Imagemagick(Package):
     depends_on('libtiff')
     depends_on('ghostscript')
 
+    def url_for_version(self, version):
+        return "https://github.com/ImageMagick/ImageMagick/archive/{0}.tar.gz".format(version)
```

Comment:
>
> We can't use http://www.imagemagick.org/download/ because as @lee218llnl
> <https://github.com/lee218llnl>discovered, it changes at least monthly.


I see...  so they keep removing old versions?

GitHub is just a mirror for GitLab. We use mirrors of plenty of packages
> instead of the primary download repository.


There are possible security issues here.  @tgamblin what is your take on
this?  Should we "go to the horse's mouth" when possible?  Or do we not
care so much?

Finally... do you think we should download using:
  a) Spack's `git` download method?
  b) Downloading a tarball from the archives/ folder?  (Which I believe was
manually created)
  c) Sending the right URL to GitHub (causing a tarball to be
auto-generated on the fly)
        https://developer.github.com/v3/repos/contents/#get-archive-link



On Thu, Aug 11, 2016 at 3:32 PM, Adam J. Stewart <notifications@github.com>
wrote:

> In var/spack/repos/builtin/packages/ImageMagick/package.py:
>
> > @@ -42,6 +43,9 @@ class Imagemagick(Package):
> >      depends_on('libtiff')
> >      depends_on('ghostscript')
> >
> > +    def url_for_version(self, version):
> > +        return "https://github.com/ImageMagick/ImageMagick/archive/{0}.tar.gz".format(version)
>
> We can't use http://www.imagemagick.org/download/ because as @lee218llnl
> <https://github.com/lee218llnl> discovered, it changes at least monthly.
> GitHub is just a mirror for GitLab. We use mirrors of plenty of packages
> instead of the primary download repository.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/1041/files/f147d645272bc96e57c707f39a80b64d1e637fb1..633e540a1718a5cc515725b13d3f1740bb950bb6#r74488003>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB1cdz7mcmWmY6VPx_9J43XlJpV0gfE_ks5qe3jogaJpZM4IzSyf>
> .
>